lib/_stream_writable.js

```diff
@@ -191,14 +190,13 @@ function validChunk(stream, state, chunk, cb) {
   if (er) {
     stream.emit('error', er);
     process.nextTick(cb, er);
-    valid = false;
+    return false;
   }
-  return valid;
+  return true;
 }
```

Comment:
It's very low level and definitely valid for some old version of V8. It is due to how v8 "compiles" this code, basically having a single return statement allow V8 to produce "faster" code.

Not sure if still applies.. @trevnorris can definitely explain it better than I do.